ocean_lib/ocean/ocean_pool.py

```diff
@@ -391,10 +409,22 @@ def add_liquidity_finalized(
         assert self._is_valid_pool(pool_address), "The pool address is not valid."
         dt_address = self.get_token_address(pool_address)
         dt = BToken(self.web3, dt_address)
-        dt.approve(pool_address, max_data_token_amount_base, from_wallet=from_wallet)
+        if dt.balanceOf(from_wallet.address) > max_data_token_amount_base:
+            raise InsufficientBalance(
+                f"Insufficient funds for adding liquidity for {dt.address} datatoken!"
+            )
+        if dt.allowance(from_wallet.address, pool_address) < max_data_token_amount_base:
+            dt.approve(
+                pool_address, max_data_token_amount_base, from_wallet=from_wallet
+            )
 
         OCEAN = BToken(self.web3, self.ocean_address)
-        OCEAN.approve(pool_address, max_OCEAN_amount_base, from_wallet=from_wallet)
+        if OCEAN.balanceOf(from_wallet.address) > max_OCEAN_amount_base:
```

Comment:
Same as previous comment, `>` is supposed to be `<`

Reply:
Sure, my bad 👍🏻